### PR TITLE
Don't use PARTITIONED join with at most scalar build side

### PR DESCRIPTION
**Pull request by kokosing**
Dont use PARTITIONED join with at most scalar build side

### CODE REVIEW

**Comment by findepi** [comment]
How this relates to https://github.com/starburstdata/presto/pull/16?

**Comment by kokosing** [comment]
> How this relates to #16?

It is just another change. However I believe there are some code conflicts.

File: presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/ReorderJoins.java [hunk 1]
```diff
@@ -377,9 +382,13 @@ private JoinEnumerationResult setJoinNodeProperties(JoinNode joinNode)
...
             if (joinDistributionType.canRepartition() && !joinNode.isCrossJoin()) {
                 JoinNode node = joinNode.withDistributionType(PARTITIONED);
-                possibleJoinNodes.add(new JoinEnumerationResult(costProvider.getCumulativeCost(node), Optional.of(node)));
+                if (!isAtMostScalar(node.getRight(), lookup)) {
+                    possibleJoinNodes.add(new JoinEnumerationResult(costProvider.getCumulativeCost(node), Optional.of(node)));
+                }
                 node = node.flipChildren();
-                possibleJoinNodes.add(new JoinEnumerationResult(costProvider.getCumulativeCost(node), Optional.of(node)));
+                if (!isAtMostScalar(node.getRight(), lookup)) {
+                    possibleJoinNodes.add(new JoinEnumerationResult(costProvider.getCumulativeCost(node), Optional.of(node)));
```
Comment (findepi) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/ReorderJoins.java:390]: I think this is not correct. I think that when the `getJoinDistributionType(session) == PARTITIONED` and we join `A` with `B` and `B` is scalar, we should generate broadcast join `A-B`. However, I think we will generate repartitioned join `B-A`.

**Review by findepi — CHANGES_REQUESTED** [review]
Please turn this into `!fixup Add ReorderJoins rule to pick the best join order` and point at epic 16.

**Comment by kokosing** [comment]
Superseded by: https://github.com/starburstdata/presto/pull/27

**Pull request closed without merging**